### ISSUE TITLE
Layering 3 plots, with 2 plots sharing the y-axis

### ISSUE THREAD

**Issue by sderuiter** [altair-viz/altair#873]
I have successfully created a layer chart with two layers and independent y-scales.

```
#base encoding the X-axis
base = alt.Chart(concatenated_data).encode(alt.X('d:O', axis=alt.Axis(title='Weeks')))

history = base.mark_bar().encode( alt.Y('sales_pw', axis=alt.Axis(title='Widgets sales per week')))
cumulative = base.mark_line(color="#ffd100", strokeWidth=5).encode(alt.Y('sales_cum', axis=alt.Axis(title='Cumulative Widget Sales')))


#Graph containing actual data only (using independent scales)
actual_date_graph = (history + cumulative).resolve_scale(y='independent')
actual_date_graph
```
*Result:*
![visualization 1](https://user-images.githubusercontent.com/30111/40270742-de3c9720-5b92-11e8-8aee-5164010b7690.png)

The next step is adding a third layer, where 2 plots have similar values and need to share the y-axis. 

Below is the graph for the 2 plots with values in the same range, sharing the y-axis (looks as expected):
```
#third plot, using same x-axis
prediction = base.mark_bar(color="#ddd").encode(alt.Y('predicted_sales_pw', axis=alt.Axis(title='Predicted Widget sales per week')))

#Graph showing weekly historical sales and weekly predicted sales
weekly_graph = (history + prediction)
weekly_graph
```
*Result:*
![visualization 2](https://user-images.githubusercontent.com/30111/40270771-534c4ac4-5b93-11e8-9d22-c9d38a040f74.png)

However, I can't get the 3-layer plot to function whereby 2 plots share the y-axis, while the third plot has an independent y-axis.
```
#Show all data, but have weekly sales share the Y-axis
all_data_graph = (weekly_graph + cumulative)
all_data_graph
```
*Result:*
![visualization 4](https://user-images.githubusercontent.com/30111/40270787-b42696b0-5b93-11e8-8a3f-507e4883af9e.png)

```
#Show all data, but have weekly sales share the Y-axis
all_data_graph = (weekly_graph + cumulative).resolve_scale(y='independent')
all_data_graph
```
*Result:*
![visualization 5](https://user-images.githubusercontent.com/30111/40270812-080316a0-5b94-11e8-8c0f-c28dd1eee8e3.png)

Question: is it possible to have 2 plots share the y-axis while adding a third plot that has an independent y-axis?

**Comment by jakevdp** [comment]
Yes: instead of ``weekly_graph + cumulated`` (which results in a single layered chart with three layers) try ``alt.layer(weekly_graph, cumulated)``, which creates a layered chart, the layers of which are the first layered chart and the third layer. THen (I believe) the resolve will only apply to the first two layers.

**Comment by gsctree** [comment]
alt.layer(df_fi_line+df_line,df_line1).resolve_scale(y='independent')
![visualization](https://user-images.githubusercontent.com/5775517/46257653-e8fe2e00-c4ef-11e8-80d1-9b3a30377e04.png)
